python/triton/language/core.py

```diff
@@ -1727,10 +1818,13 @@ def inline_asm_elementwise(asm: str, constraints: str, args: list, dtype, is_pur
             # Change the shape of each argument based on the broadcast shape
             for i, item in enumerate(dispatch_args):
                 dispatch_args[i], _ = bin_op_type_checking(item, broadcast_arg)
-            res_ty = block_type(dtype, broadcast_arg.shape)
+            res_tys = [block_type(dt, broadcast_arg.shape) for dt in dtype]
     handles = [t.handle for t in dispatch_args]
-    call = _builder.create_inline_asm(asm, constraints, handles, res_ty.to_ir(_builder), is_pure, pack)
-    return tensor(call, res_ty)
+    call = _builder.create_inline_asm(asm, constraints, handles, [ty.to_ir(_builder) for ty in res_tys], is_pure, pack)
+
+    if not has_multiple_outputs:
+        return tensor(call.get_result(0), res_tys[0])
+    return tuple(tensor(call.get_result(i), ty) for i, ty in enumerate(res_tys))
```

Comment:
I thought we could always return a tuple and it would just work?

Reply:
Wouldn't that break all existing users, who would now have to unpack the tuple?

Reply:
yes you're probably right, never mind